Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM golang as test
```

Comment:
What is the purpose of copying everything into the test layer, just to copy it over to the build layer?

Reply:
That's a good question. If I create a new PR with just the build and final stages (i.e. test and build combined into one), would you be willing to set up Docker Hub auto builds?

Reply:
sure